omod-2.2/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/resource/openmrs2_2/DrugReferenceMapResource2_2.java

```diff
@@ -0,0 +1,142 @@
+		return delegate;
+	}
+	
+	@Override
+	public Drug getParent(DrugReferenceMap instance) {
+		return instance.getDrug();
+	}
+	
+	@Override
+	public void setParent(DrugReferenceMap instance, Drug parent) {
+		instance.setDrug(parent);
+	}
+	
+	@Override
+	public PageableResult doGetAll(Drug parent, RequestContext context) throws ResponseException {
+		List<DrugReferenceMap> maps = new ArrayList<DrugReferenceMap>(parent.getDrugReferenceMaps());
+		return new NeedsPaging<DrugReferenceMap>(maps, context);
+	}
+	
+	@Override
+	public DrugReferenceMap getByUniqueId(String uniqueId) {
+		
+		return Context.getService(RestHelperService.class).getObjectByUuid(DrugReferenceMap.class, uniqueId);
+	}
+	
+	@Override
+	protected void delete(DrugReferenceMap delegate, String reason, RequestContext context) throws ResponseException {
+		throw new ResourceDoesNotSupportOperationException();
+	}
+	
+	@Override
+	public void purge(DrugReferenceMap delegate, RequestContext context) throws ResponseException {
+		throw new ResourceDoesNotSupportOperationException();
+		
+	}
+	
+	@PropertyGetter("display")
+	public String getDisplayString(DrugReferenceMap map) {
+		if (map.getConceptMapType() == null) {
+			return "";
+		}
+		return map.getConceptMapType().getDescription();
+	}
+	
+	@PropertyGetter("conceptMapTypeName")
+	public String getconceptMapTypeName(DrugReferenceMap map) {
+		String name = map.getConceptMapType().getName();
+		return name;
+	}
+	
+	@Override
+	public DelegatingResourceDescription getRepresentationDescription(Representation rep) {
+		if (rep instanceof DefaultRepresentation) {
+			DelegatingResourceDescription description = new DelegatingResourceDescription();
+			description.addProperty("display");
+			description.addProperty("uuid");
+			description.addProperty("auditInfo");
+			description.addSelfLink();
+			description.addLink("full", ".?v=" + RestConstants.REPRESENTATION_FULL);
+			return description;
+		} else if (rep instanceof FullRepresentation) {
+			DelegatingResourceDescription description = new DelegatingResourceDescription();
+			description.addProperty("display");
+			description.addProperty("uuid");
+			description.addProperty("auditInfo");
+			description.addProperty("conceptMapTypeName");
+			description.addSelfLink();
+			return description;
+		}
+		return null;
+	}
+	
+	public Model getGETModel(Representation rep) {
+		ModelImpl modelImpl = (ModelImpl) super.getGETModel(rep);
+		if (rep instanceof DefaultRepresentation) {
+			modelImpl
+			        .property("display", new StringProperty())
+			        .property("auditInfo", new StringProperty())
+			        .property("uuid", new StringProperty());
+		} else if (rep instanceof FullRepresentation) {
+			modelImpl
+			        .property("display", new StringProperty())
+			        .property("uuid", new StringProperty())
+			        .property("auditInfo", new StringProperty())
+			        .property("conceptMapTypeName", new StringProperty());
+			;
```

Comment:
semicolon on line 137 can be removed

Reply:
thanks @haripriya999 , ill updare the PR

Reply:
i updated my Pr and added tests